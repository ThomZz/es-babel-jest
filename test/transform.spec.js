import { process as esbjprocess } from '..'

describe('process', () => {
  const warn = jest.fn(console.warn)
  const oldConsole = global.console
  global.console = { ...oldConsole, warn }

  const code = `
  export const foo =  42
  export {default as noop} from 'noop3'
  export {URL} from 'url'

  jest.mock("noop3")
`;
  const file = './null.js';
  const options = {
    config: {
      cwd: process.cwd()
    }
  };

  afterAll(() => {
    global.console = oldConsole;
    jest.clearAllMocks();
  });

  it('should process', () => {
    expect(esbjprocess(code, file, options)).toMatchSnapshot();
    expect(warn.mock.calls.join('')).not.toMatch(
      /could not be resolved â€“ treating it as an external dependency/
    );
  });

  it("should always hoist mocked modules at the top",() => {
    expect(esbjprocess(code, file, options).code.startsWith('_getJestObj().mock("noop3");'));
  });

  it('should add inline snapshot', () => {
    expect(1).toMatchInlineSnapshot(`1`);
  });
});
