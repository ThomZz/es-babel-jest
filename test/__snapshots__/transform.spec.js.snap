// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`process should process 1`] = `
"var __create = Object.create;
var __defProp = Object.defineProperty;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __markAsModule = (target) => __defProp(target, \\"__esModule\\", {value: true});
var __export = (target, all) => {
  __markAsModule(target);
  for (var name in all)
    __defProp(target, name, {get: all[name], enumerable: true});
};
var __exportStar = (target, module2, desc) => {
  __markAsModule(target);
  if (module2 && typeof module2 === \\"object\\" || typeof module2 === \\"function\\") {
    for (let key of __getOwnPropNames(module2))
      if (!__hasOwnProp.call(target, key) && key !== \\"default\\")
        __defProp(target, key, {get: () => module2[key], enumerable: !(desc = __getOwnPropDesc(module2, key)) || desc.enumerable});
  }
  return target;
};
var __toModule = (module2) => {
  if (module2 && module2.__esModule)
    return module2;
  return __exportStar(__defProp(module2 != null ? __create(__getProtoOf(module2)) : {}, \\"default\\", {value: module2, enumerable: true}), module2);
};
__export(exports, {
  URL: () => url.URL,
  foo: () => foo,
  noop: () => noop3.default
});
var noop3 = __toModule(require(\\"noop3\\"));
var url = __toModule(require(\\"url\\"));
const foo = 42;
//# sourceMappingURL=data:application/json;base64,ewogICJ2ZXJzaW9uIjogMywKICAic291cmNlcyI6IFsiPHN0ZGluPiIsICI8c3RkaW4+Il0sCiAgInNvdXJjZXNDb250ZW50IjogWyJcbiAgICAgICAgZXhwb3J0IGNvbnN0IGZvbyA9ICA0MlxuICAgICAgICBleHBvcnQge2RlZmF1bHQgYXMgbm9vcH0gZnJvbSAnbm9vcDMnXG4gICAgICAgIGV4cG9ydCB7VVJMfSBmcm9tICd1cmwnXG4gICAgIiwgIlxuICAgICAgICBleHBvcnQgY29uc3QgZm9vID0gIDQyXG4gICAgICAgIGV4cG9ydCB7ZGVmYXVsdCBhcyBub29wfSBmcm9tICdub29wMydcbiAgICAgICAgZXhwb3J0IHtVUkx9IGZyb20gJ3VybCdcbiAgICAiXSwKICAibWFwcGluZ3MiOiAiOzs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7Ozs7OztBQUFBO0FBQUE7QUFBQTtBQUFBO0FBQUE7QUNFUSxJQUFBLFFBQThCO0FBQzlCLFVBQWtCO0FBRlgsWUFBYTsiLAogICJuYW1lcyI6IFtdCn0K
"
`;
